### PR TITLE
Via issues preview UI

### PR DESCRIPTION
**Pull request by via-codes**
https://github.com/bignerdranch/Github-Issues-Reader/issues/2 

Added State status logic 
Took out all storyboard references and made it exclusively programmatic with the one nib file
Created a NavController 
Issue title bold and single line and Truncate text
Issue User subtext
Person SF symbol added next to username
Issue state green when "open" red when "closed" and displayed all on the same subtext like as user. 

### Testing
Launch app
Enter organization and repository (or use the auto populated "Apple", "Swift" text)
Tap "View Issues" button
App will automatically navigate to the next screen, the Issues Preview.

### CODE REVIEW

File: Github Issue Reader/Views/IssuesCollectionVC.swift [hunk 1]
```diff
@@ -11,6 +11,10 @@ private let reuseIdentifier = "Cell"
 
 class IssuesCollectionVC: UICollectionViewController, UICollectionViewDelegateFlowLayout {
     
+    func title(){
```
Comment (kevinrandrup) [inline Github Issue Reader/Views/IssuesCollectionVC.swift]: Lil space here

File: IssuePreviewCVCell.swift [hunk 2]
```diff
@@ -11,26 +11,95 @@ import UIKit
...
+    }
     
+    func configure(issue: Issue) {
+        issueTitle.text = issue.title
+        issueUsername.text = issue.user?.login
+        issueStatus.configuration?.title = issue.state.rawValue
+        
+        issueStatus.configuration?.baseBackgroundColor = .green
+        
+        if issue.state == .open {
```
Comment (kevinrandrup) [inline IssuePreviewCVCell.swift]: Would you please use a `switch` statement instead?

File: IssuePreviewCVCell.swift [hunk 3]
```diff
@@ -11,26 +11,95 @@ import UIKit
...
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
     
+    func configure(issue: Issue) {
+        issueTitle.text = issue.title
+        issueUsername.text = issue.user?.login
+        issueStatus.configuration?.title = issue.state.rawValue
+        
+        issueStatus.configuration?.baseBackgroundColor = .green
```
Comment (kevinrandrup) [inline IssuePreviewCVCell.swift]: This property immediately gets changed below so doesn't do anything here

File: IssuePreviewView.swift [hunk 4]
```diff
@@ -34,8 +34,8 @@ class IssuePreviewView: UIView {
        
     }
     
-    class func nib() -> IssuePreviewView {
-        return UINib(nibName: "IssuePreviewView", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! IssuePreviewView
-    }
+//    class func nib() -> IssuePreviewView {
```
Comment (kevinrandrup) [inline IssuePreviewView.swift:37]: We don't need this file anymore, right?

File: Github Issue Reader/ViewController.swift [hunk 5]
```diff
@@ -126,11 +126,12 @@ class ViewController: UIViewController {
         viewModel.fetchIssues(for: organization, repo: repo) { [self] issues in
             print("Successfully received issues.")
             
-            for issue in self.viewModel.issues {
-                print(issue.title)
-            }
+//            for issue in self.viewModel.issues {
+//                print(issue.title)
```
Comment (markd2) [inline Github Issue Reader/ViewController.swift]: I try not to check in caveman debugging or development printage.  (And also try not to check in commented-out swaths of code)

File: Github Issue Reader/Views/IssuesCollectionVC.swift [hunk 6]
```diff
@@ -64,9 +62,8 @@ class IssuesCollectionVC: UICollectionViewController, UICollectionViewDelegateFl
...
     }
     
-//    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-//        return CGSize.init(width: view.frame.size.width, height: 250)
-//    }
-    
-    // collectionView data source protocols - common pattern
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let fullWidth = collectionView.bounds.size.width - collectionView.layoutMargins.left - collectionView.layoutMargins.right
+        return CGSize(width: fullWidth, height: 100)
```
Comment (markd2) [inline Github Issue Reader/Views/IssuesCollectionVC.swift]: you can make a constant for the height  (e.g. `let cellHeight: CGFloat = 100`) - that'll make it easier for someone later on to change (this really should be EIGHTY SEVEN POINTS AND NO MORE!!1!!), and if anything else needs to be 100 points tall to match the cell height

File: IssuePreviewCVCell.swift [hunk 7]
```diff
@@ -11,26 +11,95 @@ import UIKit
...
+    var containerHView: UIStackView = {
+        let view = UIStackView()
+        view.axis = .horizontal
+        view.alignment = .fill
+        view.distribution = .fill
+        view.spacing = 20
+        return view
+    }()
+    
+    var issueTitle: UILabel = {
```
Comment (markd2) [inline IssuePreviewCVCell.swift]: common nomenclature for UI components is including the type in the name, like `issueTitleLabel`.  `issueTitle` kind of sounds stringy, like this is the issue title, not the thing that is displaying the issue title.  (and this applies to things up and down here)

File: IssuePreviewCVCell.swift (hunk 3 above)
Comment (via-codes) [reply IssuePreviewCVCell.swift]: Whoops. Def was suppose to delete that.

File: IssuePreviewCVCell.swift [hunk 8]
```diff
@@ -11,26 +11,95 @@ import UIKit
...
+        var label = UILabel()
+        label.font = UIFont.preferredFont(forTextStyle: .body)
+        label.lineBreakMode = .byTruncatingTail
+        return label
+    }()
+    
+    let userPic = UIImageView (image: UIImage(systemName: "person.circle"))
```
Comment (markd2) [inline IssuePreviewCVCell.swift]: nit: can lose the space before the `(`

File: IssuePreviewView.swift (hunk 4 above)
Comment (via-codes) [reply IssuePreviewView.swift:37]: Correct! I totally didn't realize that we could nix it altogether. Shall I do that?

File: IssuePreviewCVCell.swift [hunk 9]
```diff
@@ -11,26 +11,95 @@ import UIKit
...
+    
+    var issueUsername: UILabel = {
+        var label = UILabel()
+        label.font = UIFont.preferredFont(forTextStyle: .caption1)
+        return label
+    }()
+    
+    var spacerView: UIView = {
+        let spacer = UIView()
+        // Create a view to use as a spacer which expands to fill available area
```
Comment (markd2) [inline IssuePreviewCVCell.swift:49]: oooh, I'm totally stealing this

**Review by markd2 — COMMENTED** [review]
wow, looks pretty spif!

File: IssuePreviewView.swift (hunk 4 above)
Comment (kevinrandrup) [reply IssuePreviewView.swift:37]: Yeet it

File: Github Issue Reader/ViewController.swift [hunk 10]
```diff
@@ -126,11 +126,8 @@ class ViewController: UIViewController {
         viewModel.fetchIssues(for: organization, repo: repo) { [self] issues in
             print("Successfully received issues.")
             
-            for issue in self.viewModel.issues {
-                print(issue.title)
-            }
-            
             let layout = UICollectionViewFlowLayout()
+            layout.itemSize = UICollectionViewFlowLayout.automaticSize
```
Comment (kevinrandrup) [inline Github Issue Reader/ViewController.swift]: We manually calculate the height and width of the cell so this doesn't need to be set

File: IssuePreviewView.swift (hunk 4 above)
Comment (kevinrandrup) [reply IssuePreviewView.swift:37]: Followup, this didn't get deleted?

File: Github Issue Reader/Views/IssuesCollectionVC.swift [hunk 11]
```diff
@@ -11,6 +11,12 @@ private let reuseIdentifier = "Cell"
 
 class IssuesCollectionVC: UICollectionViewController, UICollectionViewDelegateFlowLayout {
     
+    var cellHeight: CGFloat = 100
+    
+    func title() {
+        navigationItem.title = "Issues"
+    }
```
Comment (kevinrandrup) [inline Github Issue Reader/Views/IssuesCollectionVC.swift]: This function is never called so it never assigns the title of the `navigationItem` to "Issues". It's also not an override that changes an existing function.

File: Github Issue Reader/Views/IssuesCollectionVC.swift [hunk 12]
```diff
@@ -21,20 +27,14 @@ class IssuesCollectionVC: UICollectionViewController, UICollectionViewDelegateFl
...
     
-    
     func configureViewController() {
         view.backgroundColor = .systemBackground
-        navigationController?.navigationBar.prefersLargeTitles = true
-        
-        if let flowLayout = collectionView?.collectionViewLayout as? UICollectionViewFlowLayout {
-            flowLayout.estimatedItemSize = UICollectionViewFlowLayout.automaticSize
-        }
+        navigationController?.navigationBar.prefersLargeTitles = true
```
Comment (kevinrandrup) [inline Github Issue Reader/Views/IssuesCollectionVC.swift:34]: The navigation bar isn't visible, so this doesn't so anything

**Review by kevinrandrup — CHANGES_REQUESTED** [review]
Missing the first AC in #2

> Large title "Issue" that moves into the navigation bar when scrolling. This functionality is built into UINavigationController.

**Review by kevinrandrup — APPROVED** [review]

**Pull request merged**